R/http-requests.R

```diff
@@ -65,3 +65,54 @@ gsheets_GET <- function(url) {
   req$content <- XML::xmlToList(req$content)
   req
 }
+
+
+
+#' Create POST request
+#'
+#' Make POST request to Google Sheets/Drive API.
+#'
+#' @param url URL for POST request
+#' @param the_body body of POST request
+gsheets_POST <- function(url, the_body)
+{
+  token <- get_google_token()
+  
+  if(is.null(token)) {
+    stop("Must be authorized in order to perform request")
+  } else {
+    
+    # first look at the url to determine contents, 
+    # must be either talking to "drive" or "spreadsheets" API
+    if(stringr::str_detect(stringr::fixed(url), "drive")) {
+      content_type <- "application/json" # send json to drive api
```

Comment:
I note that `httr::POST()` has the argument `encode =` which could be used to JSON encode here, instead of pre-JSONifying as we do in, e.g. `new_ss()`. In general, we're trying to use any `httr` smarts we can and only do stuff ourselves if we must. But maybe @jzhaoo did this on purpose because it doesn't look like `httr` is willing to encode stuff as XML, which apparently we need to do. So then things wouldn't be a parallel as they currently are for the Drive API vs the Sheets API in this function. I put the comment here because I'd guess we don't need to set content type explicitly if we allow `httr::POST()` to do the encoding for us.